src/adapter/src/coord.rs

```diff
@@ -1351,6 +1355,36 @@ impl Coordinator {
             let _ = meta.notice_tx.send(notice.clone());
         }
     }
+
+    fn new_temporary_dataflow(&mut self, cluster: ClusterId) -> Result<(), AdapterError> {
+        let value = self
+            .active_dataflows_per_cluster
+            .get(&cluster)
+            .unwrap_or(&0);
+        self.validate_resource_limit(
+            *value,
+            1,
+            SystemVars::max_temporary_dataflows_per_cluster,
+            "temporary dataflows",
+            "max_temporary_dataflows_per_cluster",
```

Comment:
I think that having temporary dataflows as user facing concept will end up in a lot of confusion. Is there some other concept we can use here? Maybe for now we should just limit the number of subscribes and not consider peeks.